lib/Dialect/Ibis/IbisOps.cpp

```diff
@@ -16,6 +18,123 @@ using namespace mlir;
 using namespace circt;
 using namespace ibis;
 
+ParseResult FuncOp::parse(OpAsmParser &parser, OperationState &result) {
+  // Parse the name as a symbol.
+  StringAttr nameAttr;
+  if (parser.parseSymbolName(nameAttr, SymbolTable::getSymbolAttrName(),
+                             result.attributes))
+    return failure();
+
+  // Parse the function signature.
+  SmallVector<OpAsmParser::Argument, 4> args;
+  SmallVector<Attribute> argNames;
+  SmallVector<Type> resultTypes;
+  TypeAttr functionType;
+
+  using namespace mlir::function_interface_impl;
+  auto *context = parser.getContext();
+
+  // Parse the argument list.
+  if (parser.parseArgumentList(args, OpAsmParser::Delimiter::Paren,
+                               /*allowType=*/true, /*allowAttrs=*/false))
+    return failure();
+
+  // Parse the result type.
+  if (succeeded(parser.parseOptionalArrow())) {
+    Type resultType;
+    if (parser.parseType(resultType))
+      return failure();
+    resultTypes.push_back(resultType);
+  }
+
+  // Process the ssa args for the information we're looking for.
+  SmallVector<Type> argTypes;
+  for (auto &arg : args) {
+    argNames.push_back(
+        hw::module_like_impl::getPortNameAttr(context, arg.ssaName.name));
```

Comment:
Surprised to see `hw` here - are you using this strictly out of laziness? If not, then i'd expected an Ibis func to actually be `HWModuleLike` - ... which it is not. 

Given that, i don't think it's good style to have `hw` stuff here strictly for using a 9-line helper function.

Reply:
Laziness? Dunno about that. Code sharing good! Given that everything has a dependence on HW, I'm ok with it. There is an argument, however, for pulling this function out and putting it in the support lib since it is fairly generic functionality...

Reply:
#5657